### ISSUE TITLE
Safari - Challenge links on "Challenge" page  do not work

### ISSUE THREAD

**Issue by zoomequipd** [CTFd/CTFd#556]
**Environment**:

  - CTFd Version/Commit: 
  - Operating System: Mac OS X - Version 10.12.6
  - Web Browser and Version:  Safari Version 11.0.2

**What happened?**
On Safari - the modals for some challenge links do not work
On Chrome - All modals for challenge links work.

**What did you expect to happen?**
Functional parity between Chrome and Safari

**How to reproduce your issue**
As an admin - 
  Create a challenge - name it X
  Access the "Challenge" page
  on the X challenge, click the icon representing "preview X"
  ---- no modal comes up
  on the X challenge, click the icon representing "edit X"
  ---- modal comes up

**Any associated stack traces or error logs**
No

**Comment by ColdHeat** [comment]
My suspicion is that FontAwesome is leveraging the same handler (document.ready) that CTFd is using. Then on top of that Safari must be choosing to execute the document.ready handler asynchronously because occasionally the handler works. 

The fix for this is to move the fontawesome script to after the body. 

```
diff --git a/CTFd/themes/admin/templates/base.html b/CTFd/themes/admin/templates/base.html
index cc94b6b..b6aa721 100644
--- a/CTFd/themes/admin/templates/base.html
+++ b/CTFd/themes/admin/templates/base.html
@@ -16,7 +16,6 @@
        <script src="{{ request.script_root }}/themes/admin/static/js/vendor/moment.min.js"></script>
        <script src="{{ request.script_root }}/themes/admin/static/js/vendor/moment-timezone-with-data.min.js"></script>
        <script src="{{ request.script_root }}/themes/admin/static/js/vendor/nunjucks.min.js"></script>
-       <script src="{{ request.script_root }}/themes/admin/static/js/vendor/fontawesome-all.min.js"></script>
        <script type="text/javascript">
                        var script_root = "{{ request.script_root }}";
        </script>
@@ -115,6 +114,7 @@
        <script src="{{ request.script_root }}/themes/admin/static/js/vendor/jquery.min.js"></script>
        <script src="{{ request.script_root }}/themes/admin/static/js/vendor/marked.min.js"></script>
        <script src="{{ request.script_root }}/themes/admin/static/js/vendor/bootstrap.bundle.min.js"></script>
+       <script src="{{ request.script_root }}/themes/admin/static/js/vendor/fontawesome-all.min.js"></script>
        <script src="{{ request.script_root }}/themes/admin/static/js/main.js"></script>
        <script src="{{ request.script_root }}/themes/admin/static/js/utils.js"></script>
        <script src="{{ request.script_root }}/themes/admin/static/js/ezq.js"></script>
```

Perhaps other scripts need to be moved after the body as well.

**Comment by ColdHeat** [comment]
For some reason, this bug seems more deeply rooted in Safari. I don't understand why but Safari specifically is not processing document.ready handlers properly. 

Needs some more research so I'll probably bundle this in with the updates to Bootstrap.

**Comment by ColdHeat** [comment]
The real issue is some kind of timing issue that doesn't manifest itself in Chrome or other sane browsers...

For some reason Safari decides to load font awesome async or something to that extent. Then when the browser is getting to the document.ready handlers it's possible for the icons to not be ready. Because the new font awesome replaces the original icon tag with an svg icon, if the event handler registration occurs too early the button won't work. 

So to fix it we just make a wrapper element that font awesome won't try to modify. Only question is are there other situations where this exists.

**Comment by ColdHeat** [comment]
Closed in #558
